contract-examples/contracts/ExampleFeeManager.sol

```diff
@@ -84,3 +104,131 @@ contract ExampleFeeManager is AllowList {
+
+  function test_addContractToManagerList() public {
+    ExampleFeeManager example = new ExampleFeeManager();
+
+    address exampleAddress = address(example);
+    address thisAddress = address(this);
+
+    IFeeManager manager = IFeeManager(FEE_MANAGER_ADDRESS);
+
+    // TODO: make this a const (role: ADMIN = 2)
+    assertEq(manager.readAllowList(thisAddress), 2);
+    assertEq(manager.readAllowList(exampleAddress), 0);
+
+    manager.setEnabled(exampleAddress);
+
+    // TODO: make this a const (role: ENABLED = 1)
+    assertEq(manager.readAllowList(exampleAddress), 1);
+  }
+
+  function test_changeFees() public {
+    ExampleFeeManager example = new ExampleFeeManager();
+    address exampleAddress = address(example);
+
+    IFeeManager manager = IFeeManager(FEE_MANAGER_ADDRESS);
+
+    manager.setEnabled(exampleAddress);
+
+    FeeConfig memory config = example.getCurrentFeeConfig();
+
+    FeeConfig memory newFeeConfig = FeeConfig({
+      gasLimit: CCHAIN_GAS_LIMIT,
+      targetBlockRate: CCHAIN_TARGET_BLOCK_RATE,
+      minBaseFee: CCHAIN_MIN_BASE_FEE,
+      targetGas: CCHAIN_TARGET_GAS,
+      baseFeeChangeDenominator: CCHAIN_BASE_FEE_CHANGE_DENOMINATOR,
+      minBlockGasCost: CCHAIN_MIN_BLOCK_GAS_COST,
+      maxBlockGasCost: CCHAIN_MAX_BLOCK_GAS_COST,
+      blockGasCostStep: CCHAIN_BLOCK_GAS_COST_STEP
+    });
+
+    assertNotEq(config.gasLimit, newFeeConfig.gasLimit);
+    // target block rate is the same for wagmi and cchain
+    // assertNotEq(config.targetBlockRate, newFeeConfig.targetBlockRate);
+    assertNotEq(config.minBaseFee, newFeeConfig.minBaseFee);
+    assertNotEq(config.targetGas, newFeeConfig.targetGas);
+    assertNotEq(config.baseFeeChangeDenominator, newFeeConfig.baseFeeChangeDenominator);
+    // min block gas cost is the same for wagmi and cchain
+    // assertNotEq(config.minBlockGasCost, newFeeConfig.minBlockGasCost);
+    assertNotEq(config.maxBlockGasCost, newFeeConfig.maxBlockGasCost);
+    assertNotEq(config.blockGasCostStep, newFeeConfig.blockGasCostStep);
+
+    example.enableCChainFees();
+
+    FeeConfig memory changedFeeConfig = example.getCurrentFeeConfig();
+
+    assertEq(changedFeeConfig.gasLimit, newFeeConfig.gasLimit);
+    // target block rate is the same for wagmi and cchain
+    // assertEq(changedFeeConfig.targetBlockRate, newFeeConfig.targetBlockRate);
+    assertEq(changedFeeConfig.minBaseFee, newFeeConfig.minBaseFee);
+    assertEq(changedFeeConfig.targetGas, newFeeConfig.targetGas);
+    assertEq(changedFeeConfig.baseFeeChangeDenominator, newFeeConfig.baseFeeChangeDenominator);
+    // min block gas cost is the same for wagmi and cchain
+    // assertEq(changedFeeConfig.minBlockGasCost, newFeeConfig.minBlockGasCost);
+    assertEq(changedFeeConfig.maxBlockGasCost, newFeeConfig.maxBlockGasCost);
+    assertEq(changedFeeConfig.blockGasCostStep, newFeeConfig.blockGasCostStep);
+
+    assertEq(example.getFeeConfigLastChangedAt(), block.number);
+
+    example.enableCustomFees(config);
```

Comment:
is this done as a cleanup (should we add a comment or use a tearDown fn, not sure if that exists?)

Reply:
There's no tear-down function, but we could add one. This is literally the only place where I had to do a cleanup, so I didn't want to add a teardown but I think it's a good idea.